### PR TITLE
Add row and column directives 

### PR DESCRIPTION
**Pull request by ethan-kusters**
Bug/issue #, if applicable: rdar://97739637 

## Summary

Adds support for `@Row` and `@Column` in reference documentation pages.

Described on the forums here: https://forums.swift.org/t/supporting-more-dynamic-content-in-swift-docc-reference-documentation/59527#row-9.

## Dependencies

- https://github.com/apple/swift-docc-render/pull/409

### CODE REVIEW

**Comment by ethan-kusters** [comment]
@swift-ci please test

**Comment by ethan-kusters** [comment]
@swift-ci please test

**Comment by ethan-kusters** [comment]
@swift-ci please test

File: Sources/SwiftDocC/Model/Rendering/Content/RenderBlockContent.swift [hunk 1]
```diff
@@ -60,6 +60,8 @@ public enum RenderBlockContent: Equatable {
     case termList(TermList)
     /// A table that contains a list of row data.
     case table(Table)
+    
+    case row(Row)
```
Comment (franklinsch) [inline Sources/SwiftDocC/Model/Rendering/Content/RenderBlockContent.swift:64]: Looks like we're missing docs here, same for the public APIs below

File: Sources/SwiftDocC/Semantics/Reference/Row.swift [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    
+    override var children: [Semantic] {
+        return columns
+    }
+    
+    var childMarkup: [Markup] {
+        return columns.flatMap(\.childMarkup)
+    }
+    
+    @available(*, deprecated,
```
Comment (franklinsch) [inline Sources/SwiftDocC/Semantics/Reference/Row.swift:40]: Is it possible to use obsoleted here?

**Review by franklinsch — APPROVED** [review]
Looks great! Just a few suggestions

File: Sources/SwiftDocC/Semantics/Reference/Row.swift (hunk 2 above)
Comment (ethan-kusters) [reply Sources/SwiftDocC/Semantics/Reference/Row.swift:40]: Unfortunately not. 🫤

Obsoleted requires a version number: https://docs.swift.org/swift-book/ReferenceManual/Attributes.html.

File: Sources/SwiftDocC/Model/Rendering/Content/RenderBlockContent.swift (hunk 1 above)
Comment (ethan-kusters) [reply Sources/SwiftDocC/Model/Rendering/Content/RenderBlockContent.swift:64]: Thanks! Addressed here: https://github.com/apple/swift-docc/pull/378/commits/d0c2e31d47fd60217994704c71566e13ea48744e.

**Comment by ethan-kusters** [comment]
@swift-ci please test

**Pull request merged**